eclair-core/src/test/scala/fr/acinq/eclair/integration/IntegrationSpec.scala

```diff
@@ -863,38 +840,79 @@ class IntegrationSpec extends TestKitBaseClass with BitcoindService with AnyFunS
     assert(outgoingPayments.forall(p => p.status.isInstanceOf[OutgoingPaymentStatus.Failed]), outgoingPayments)
   }
 
-  test("send payments and close the channel C -> F6 with option_anchor_outputs/option_static_remotekey") {
-    // initially all the balance is on C side and F6 doesn't have an output
+  test("close channel C <-> F1") {
     val sender = TestProbe()
-    sender.send(nodes("F6").register, Symbol("channelsTo"))
-    // retrieve the channelId of C <--> F6
+    sender.send(nodes("F1").register, Symbol("channelsTo"))
+    val channels = sender.expectMsgType[Map[ByteVector32, PublicKey]]
+    assert(channels.size === 1)
+    assert(channels.head._2 === nodes("C").nodeParams.nodeId)
+    val channelId = channels.head._1
+
+    sender.send(nodes("C").register, Forward(channelId, CMD_GETSTATEDATA))
+    val commitmentsC = sender.expectMsgType[DATA_NORMAL].commitments
+    val finalPubKeyScriptC = commitmentsC.localParams.defaultFinalScriptPubKey
+    val fundingOutpoint = commitmentsC.commitInput.outPoint
+    sender.send(nodes("F1").register, Forward(channelId, CMD_GETSTATEDATA))
+    val finalPubKeyScriptF = sender.expectMsgType[DATA_NORMAL].commitments.localParams.defaultFinalScriptPubKey
+
+    sender.send(nodes("F1").register, Forward(channelId, CMD_CLOSE(Some(finalPubKeyScriptF))))
+    sender.expectMsg(ChannelCommandResponse.Ok)
+    // we then wait for C and F to negotiate the closing fee
+    awaitCond({
+      sender.send(nodes("C").register, Forward(channelId, CMD_GETSTATE))
+      sender.expectMsgType[State] == CLOSING
+    }, max = 20 seconds, interval = 1 second)
+
+    generateBlocks(bitcoincli, 2)
+    awaitCond({
+      sender.send(nodes("C").register, Forward(channelId, CMD_GETSTATE))
+      sender.expectMsgType[State] == CLOSED
+    }, max = 20 seconds, interval = 1 second)
+
+    val bitcoinClient = new ExtendedBitcoinClient(bitcoinrpcclient)
+    bitcoinClient.lookForSpendingTx(None, fundingOutpoint.txid, fundingOutpoint.index.toInt).pipeTo(sender.ref)
+    val closingTx = sender.expectMsgType[Transaction]
+    assert(closingTx.txOut.map(_.publicKeyScript).toSet === Set(finalPubKeyScriptC, finalPubKeyScriptF))
+
+    awaitAnnouncements(nodes.filterKeys(_ == "A").toMap, 5, 7, 16)
+  }
+
+  test("open channel C <-> F2, send payments and close (option_anchor_outputs, option_static_remotekey)") {
```

Comment:
Do I understand correctly that you are introducing a new naming convention that F1 = standard commitment and F2 = anchor output? If so, I'm not sure that's a good idea because `C` doesn't have the same naming convention.

Reply:
It's not necessarily a naming convention, just two test nodes that support different commitment formats (just like we previously had one of the F nodes support trampoline and not the others).
I agree this file is a a bit complex to follow, especially since we want to batch some things to avoid making the tests unnecessarily slow :/